xwiki-platform-core/xwiki-platform-attachment/xwiki-platform-attachment-api/src/main/java/org/xwiki/attachment/internal/listener/MovedAttachmentListener.java

```diff
@@ -94,21 +96,22 @@ public void onEvent(Event event, Object source, Object data)
                 || this.authorization.hasAccess(Right.EDIT, moveAttachmentRequest.getUserReference(), reference))
                 && (!moveAttachmentRequest.isCheckAuthorRights() || this.authorization.hasAccess(Right.EDIT,
                     moveAttachmentRequest.getAuthorReference(), reference)));
-            // TODO: Make sure to apply the refactoring at farm level for attachments as part of XWIKI-8346.
-            updateBackLinks(attachmentMovedEvent, canEdit);
+            try {
+                updateBackLinks(attachmentMovedEvent, canEdit);
+            } catch (RefactoringException e) {
+                this.logger.error("Failed to update links backlinks", e);
```

Comment:
Would it be interesting to log some variables from the context in addition to the exception? I'm thinking of `attachmentMovedEvent` or `moveAttachmentRequest`.

Reply:
Cannot hurt, I guess. Now the cause of this exception is supposed to be that it failed to load the backlinks for that attachment reference and the cause exception contains the reference.